odoo/fields.py

```diff
@@ -1280,23 +1280,22 @@ def column_type(self):
         # with all significant digits.
         # FLOAT8 type is still the default when there is no precision because it
         # is faster for most operations (sums, etc.)
-        return ('numeric', 'numeric') if self.digits is not None else \
+        return ('numeric', 'numeric') if self._digits is not None else \
                ('float8', 'double precision')
 
-    @property
-    def digits(self):
-        if callable(self._digits):
-            with LazyCursor() as cr:
-                return self._digits(cr)
+    def get_digits(self, record):
```

Comment:
I would suggest to take an `env` as parameter instead of a `record`. It would make the API easier to call in all situations, even when there is no record (yet). The direct calls will be trivial to adjust (param `record` -> `record.env`), and it should also avoid abusive/confusing cases where a record from an unrelated model is passed.